rules/private/intellij_compiler_xml.bzl

```diff
@@ -0,0 +1,49 @@
+load(":constants.bzl", "GENERATED_SOURCES_SUBDIR", "GENERATED_TEST_SOURCES_SUBDIR")
+load(":intellij_iml.bzl", "iml_info_provider") # see https://bazel.build/designs/skylark/declared-providers.html
+
+def _impl(ctx):
+    """TODO"""
+
+    args = []
+
+    for dep in ctx.attr.iml_target_to_annotation_profile.keys():
+        args.append("--module-to-profile-mapping")
+        args.append("%s=%s" % (dep.label.name, ctx.attr.iml_target_to_annotation_profile[dep]))
+
+        # TODO: need motivating use case for test annotations. can/should test/compile be handled generically?
+
+        # examine all dependencies of the module for annotation processors
+        # (which come from dependencies on java_plugin's),
+        # collect these up.
+        all_annotation_processors = depset()
+        for compile_lib_dep in (dep[iml_info_provider].compile_lib_deps): # TODO: include transitive module compile libs?
+            all_annotation_processors += compile_lib_dep.java.annotation_processing.processor_classnames
```

Comment:
how did you figure out there was a method annotation_processing?

Reply:
OMG. SO. MUCH. PRINTF. DEBUGGING.

Accompanied by semi-fanatical will to understand this problem.

One day we will both get wise and just read the relevant proto defs from within the bazel project.